MAUI/DataForm/Editors.md

```diff
@@ -0,0 +1,961 @@
+            list.Add("United States");
+            autoCompleteItem.ItemsSource = list;
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+#### Loading complex type property values in autocomplete editor
+
+You can display the complex type property values in autocomplete editor by using [GetSource]`Link` method of the IDataFormSourceProvider class, which is used to get source list as complex property values for autocomplete editor and set it to the `IDataFormSourceProvider` property of SfDataForm.Use the `GenerateDataFormItem `event to set [DisplayMemberPath] and [SelectedValuePath] property values of AutoComplete for complex type property.
+
+N> Class cannot be directly set as data type for autocomplete editor in this complex type scenario.
+
+{% tabs %}
+{% highlight MainPage.xaml.cs %}
+
+    dataForm.ItemsSourceProvider = new DataFormItemsSourceProvider();
+    dataForm.DataObject = new DataModel();
+    dataForm.RegisterEditor("City", DataFormEditorType.AutoComplete);
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+ 
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null && e.DataFormItem.FieldName == "City" && e.DataFormItem is DataFormAutoCompleteItem autoCompleteItem)
+        {
+            autoCompleteItem.DisplayMemberPath = "City";
+            autoCompleteItem.SelectedValuePath = "PostalCode";
+        }
+    } 
+ 
+    public class DataFormItemsSourceProvider : IDataFormSourceProvider
+    {
+         public object GetSource(string sourceName)
+        {
+            if (sourceName == "City")
+            {
+                List<Address> details = new List<Address>();
+                details.Add(new Address() { City = "Colorado Springs", PostalCode = 1 });
+                details.Add(new Address() { City = "Chicago", PostalCode = 2 });
+                details.Add(new Address() { City = "Columbus", PostalCode = 3 });
+                details.Add(new Address() { City = "Portland", PostalCode = 4 });
+                details.Add(new Address() { City = "Paris", PostalCode = 5 });
+                details.Add(new Address() { City = "Las Vegas", PostalCode = 6 });         
+                details.Add(new Address() { City = "New York", PostalCode = 7 });
+                details.Add(new Address() { City = "Cincinnati", PostalCode = 8 });
+                details.Add(new Address() { City = "San Diego", PostalCode = 9 });
+
+                return details;
+            }
+            return new List<string>();
+        }
+    }
+
+    public class DataModel
+    {
+        [Display(Name ="First Name")]
+        public String FirstName { get; set; } 
+        public string City { get; set; }
+    }
+
+    public class Address
+    {
+        public int PostalCode { get; set; }
+        public string City { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+`Image`
+
+#### AutoComplete editor modes
```

Comment:
AutoComplete editor modes => remove this

Reply:
removed